drivers/kscan/Kconfig.xpt2046

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2022 Sen(1065402494@qq.com)
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig KSCAN_XPT2046
+	bool "XPT2046 touch screen controller driver"
+	depends on SPI
```

Comment:
add `depends on DT_HAS_XPT_XPT2046_ENABLED`
change `depends on SPI` to `select SPI`

Reply:
1. According to the described additions and changes, the XPT2046 driver will not work, which may be related to the undefined "DT_HAS_XPT_XPT2046_ENABLED".
2. Looking at the content in zephyr, it seems that most of the Kconfig files are written in the style submitted in the original text (because I refer to them), for example, refer to "Kconfig.ft5336".